lib/semian.rb

```diff
@@ -216,7 +218,7 @@ def unregister_all_resources
 
   # Retrieves a hash of all registered resources.
   def resources
-    @resources ||= {}
+    @resources ||= LRUCache.new(5)
```

Comment:
Setting a max size is good to start but we should also investigate an option that lets this grow unbounded and only pulls resources off the LRU if their error windows are empty.